### PR TITLE
added schedule model and crud view

### PR DESCRIPTION
**Pull request by KaitCrawford**
review and merge https://github.com/praekelt/mentorship/pull/4 first

### CODE REVIEW

File: mentorship/mentor/views.py [hunk 1]
```diff
@@ -1,12 +1,123 @@
...
+        call = Call.objects.create(
+            caller=user.contact_details,
+            receiver=user.contact_details.mentor.mentee)
+        call.dial()
+        request.session['call_id'] = call.id
+    return HttpResponseRedirect(reverse('home'))
+
+
+@login_required(login_url='/login/')
+def schedule(request):
```
Comment (miltontony) [inline mentorship/mentor/views.py]: You should use [class based generic views](https://docs.djangoproject.com/en/1.8/topics/class-based-views/intro/) for all the schedule views. You can see them in action [here](https://github.com/praekelt/mc2/blob/develop/controllers/base/views.py#L75-L111)

**Comment by KaitCrawford** [comment]
:+1: from me @miltontony?

File: mentorship/mentor/models.py [hunk 2]
```diff
@@ -50,3 +50,21 @@ class Message(models.Model):
...
+    call_time = models.DateTimeField(blank=False,
+                                     null=False)
+    caller = models.ForeignKey(Mentor)
+    callee = models.ForeignKey(Mentee)
+
+    def notify_callee(self):
+        message = Message.objects.create(
+            sender=self.caller, recipient=self.callee,
+            content="Next call scheduled")
+        message.save()
```
Comment (miltontony) [inline mentorship/mentor/models.py]: not necessary here since "objects.create()" saves it the first time.

File: mentorship/mentor/views.py [hunk 3]
```diff
@@ -66,3 +71,63 @@ def start_call(request):
         call.dial()
         request.session['call_id'] = call.id
     return HttpResponseRedirect(reverse('home'))
+
+
+@login_required(login_url='/login/')
+def schedule(request):
```
Comment (miltontony) [inline mentorship/mentor/views.py]: Why do you need this if it's the same thing as `schedule/create`? see comment on `scheduled_calls` below

File: mentorship/mentor/views.py [hunk 4]
```diff
@@ -66,3 +71,63 @@ def start_call(request):
...
+    context = {
+        'user': user,
+        'form': form
+    }
+
+    scheduled_calls = ScheduledCall.objects.filter(
+        caller=user.contact_details).exclude(
+        call_time__lte=datetime.date.today())
+    context.update({
+        'scheduled_calls': scheduled_calls,
```
Comment (miltontony) [inline mentorship/mentor/views.py]: If you move this to the `ScheduledCallCreate` (as suggested above), you can add this info using the `get_context_data()` method of the view

**Comment by KaitCrawford** [comment]
@miltontony ready again :)

File: mentorship/mentor/views.py [hunk 5]
```diff
@@ -66,3 +68,55 @@ def start_call(request):
         call.dial()
         request.session['call_id'] = call.id
     return HttpResponseRedirect(reverse('home'))
+
+
+class ScheduledCallCreate(CreateView):
+    model = ScheduledCall
+    fields = ['call_time']
+    template_name = 'mentor/schedule.html'
+    success_url = '/schedule'
```
Comment (miltontony) [inline mentorship/mentor/views.py]: You should always have a trailing slash (Django will automatically add it, but that's means an extra unnecessary step) e.g `success_url = '/schedule/'`

**Comment by KaitCrawford** [comment]
@miltontony ready for rerererereview :P

**Comment by miltontony** [comment]
:+1: :100: :cake: 
sorry for all the rework

**Comment by KaitCrawford** [comment]
no worries :D it's how we learn

**Pull request merged**